app/kubemci/pkg/gcp/urlmap/urlmapsyncer_test.go

```diff
@@ -150,3 +159,218 @@ func TestDeleteURLMap(t *testing.T) {
+			t.Errorf("unexpected list of load balancers, expected: %v, actual: %v", c.expectedList, list)
+		}
+		for _, rule := range list {
+			expectedRule, ok := c.expectedList[rule.LoadBalancerName]
+			if !ok {
+				t.Errorf("unexpected rule %v. expected rules: %v, actual: %v", rule, c.expectedList, list)
+			}
+			if rule.LoadBalancerName != expectedRule.LoadBalancerName {
+				t.Errorf("Unexpected name. expected: %s, actual: %s", expectedRule.LoadBalancerName, rule.LoadBalancerName)
+			}
+			if rule.IPAddress != expectedRule.IPAddress {
+				t.Errorf("Unexpected IP addr. expected %s, actual: %s", expectedRule.IPAddress, rule.IPAddress)
+			}
+			actualClusters := goutils.MapFromSlice(rule.Clusters)
+			for _, expected := range expectedRule.Clusters {
+				if _, ok := actualClusters[expected]; !ok {
+					t.Errorf("expected cluster %s is missing from resultant cluster list %v", expected, rule.Clusters)
+				}
+			}
+		}
+	}
+}
+
+func TestGetLoadBalancerStatus(t *testing.T) {
+	lbName := "lb-name"
+	ipAddr := "1.2.3.4"
+	clusters := []string{"cluster1", "cluster2"}
+	svcName := "svcName"
+	svcPort := "svcPort"
+	ing := &v1beta1.Ingress{
+		Spec: v1beta1.IngressSpec{
+			Backend: &v1beta1.IngressBackend{
+				ServiceName: svcName,
+				ServicePort: intstr.FromString(svcPort),
+			},
+		},
+	}
+	beMap := backendservice.BackendServicesMap{
+		svcName: &compute.BackendService{},
+	}
+
+	frp := ingresslb.NewFakeLoadBalancers("" /*name*/, nil /*namer*/)
+	namer := utilsnamer.NewNamer("mci1", lbName)
+	ums := NewURLMapSyncer(namer, frp)
+	typedUMS := ums.(*URLMapSyncer)
+
+	// Should return http.StatusNotFound when the url map does not exist.
+	status, err := ums.GetLoadBalancerStatus(lbName)
+	if !utils.IsHTTPErrorCode(err, http.StatusNotFound) {
+		t.Fatalf("expected error with status code http.StatusNotFound, found: %v", err)
+	}
+
+	// Should return the status as expected when the url map exists with status.
+	if _, err := ums.EnsureURLMap(lbName, ipAddr, clusters, ing, beMap, false /*force*/); err != nil {
+		t.Errorf("expected no error in ensuring url map, actual: %v", err)
+	}
+	status, err = ums.GetLoadBalancerStatus(lbName)
+	if err != nil {
+		t.Errorf("unexpected error in getting status description for load balancer %s, expected err = nil, actual err: %s", lbName, err)
+	}
+
+	// Verify that status description has the expected values set.
+	if status.LoadBalancerName != lbName {
+		t.Errorf("unexpected load balancer name, expected: %s, got: %s", lbName, status.LoadBalancerName)
+	}
+	if !reflect.DeepEqual(status.Clusters, clusters) {
+		t.Errorf("unexpected list of clusters, expected: %v, got: %v", clusters, status.Clusters)
+	}
+	if status.IPAddress != ipAddr {
+		t.Errorf("unpexpected ip address, expected: %s, got: %s", status.IPAddress, ipAddr)
+	}
+
+	// Remove the status from url map's description to simulate old url maps that have status on forwarding rule.
+	// It should return a non-nil error in this case.
+	name := namer.URLMapName()
```

Comment:
`removeStatus()` function below could be used here?

Reply:
done